src/components/dataTable/DataTable.tsx

```diff
@@ -0,0 +1,50 @@
+import { useFetch } from "../../utils/useFetch";
+import { DataTableRow, ID, UserName } from "./DataTableRow";
+
+export type User = {
+  id: ID;
+  name: UserName;
+  email: string;
+};
+
+export type UserResults = {
+  results: User[];
+};
+
+export function DataTable() {
+  const { data, isLoading, isError } = useFetch<UserResults>(
+    "https://randomuser.me/api/?results=50"
+  );
+
+  if (isError) {
+    throw Error("Could not load table");
```

Comment:
in this case, we want to display some error message on the screen / alert

Reply:
It currently redirects to an errorElement using react-router. Is that ok or do something else?

Reply:
I'm not familiar with this practice, did u read about it somewhere?
anyway, we're using ErrorBoundary to catch and display errors.

Reply:
those are 2 different practices.
while using react query we usually just rendering the fallback error component